VideoPlayerVideo: Need to clear pictures before switching decoder

popcornmix opened the pull request:
The new decoder may not know how clear a picture initialised from a different decoder.
For example MMALCodec wants Release() called on clear, but VideoCodecFFMPEG does not.

popcornmix commented:
ping @FernetMenta

FernetMenta commented on xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp:
```diff
@@ -179,8 +179,11 @@ void CVideoPlayerVideo::OpenStream(CDVDStreamInfo &hint, CDVDVideoCodec* codec)
   if (m_pVideoCodec)
+  {
+    m_pVideoCodec->ClearPicture(&m_picture);
+    m_pVideoCodec->Dispose();
```
please call Dispose() in destructor

other changes look ok

popcornmix commented:
@FernetMenta updated

FernetMenta commented on xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp:
```diff
@@ -99,6 +99,8 @@ CVideoPlayerVideo::CVideoPlayerVideo(CDVDClock* pClock
 CVideoPlayerVideo::~CVideoPlayerVideo()
 {
+  if (m_pVideoCodec)
+    m_pVideoCodec->Dispose();
```
Sorry, I meant the destructor of the codec. Then we can drop Dispose() from the interface of VideoCodec.

popcornmix committed:
DVDVideoCodec: Remove Dispose from API

Codecs already call this from their destructors

popcornmix committed:
VideoPlayerVideo: Need to clear pictures before switching decoder

The new decoder may not know how clear a picture initialised from a different decoder.
For example MMALCodec wants Release() called on clear, but VideoCodecFFMPEG does not.

popcornmix commented:
@FernetMenta updated

FernetMenta commented:
+1, thanks

popcornmix commented:
jenkins build this please
